src/dropdown/Dropdown.js

```diff
@@ -0,0 +1,188 @@
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import {
+  View,
+  Text,
+  TouchableNativeFeedback,
+  TouchableOpacity,
+  ActivityIndicator,
+  Platform,
+  StyleSheet,
+  ViewPropTypes,
+  TextInput,
+  AppState,
+} from 'react-native';
+import Color from 'color';
+
+import { withTheme } from '../config';
+import { renderNode, nodeType, conditionalStyle, color } from '../helpers';
+import Icon from '../icons/Icon';
+import Input from '../../'
+import { abs } from 'react-native-reanimated';
+
+const defaultLoadingProps = (type, theme) => ({
+  color: type === 'solid' ? 'white' : theme.colors.primary,
+  size: 'small',
+});
+
+const Dropdown = (props) => {
+  const { data, onItemSelected, itemSelected, containerStyle, innerStyle, containerListStyle, listStyle } = props
+  let { indexSelected } = props
+  const defaultSelected = data[0]
+  const label = typeof defaultSelected.label !== 'undefined' ? defaultSelected.label : defaultSelected.value
+  const [isListShown, setIsListShown] = React.useState(false)
+  const [dataDropdown, setDataDropdown] = React.useState([])
+  const tmpItemSelected = typeof itemSelected === 'undefined' ? null : itemSelected
+  const [selectedItem, setSelectedItem] = React.useState(tmpItemSelected)
+  const input = React.useRef(null)
+  const [inputHeigh, setInputHeigh] = React.useState(0)
+  const [index, setIndex] = React.useState(indexSelected)
+
+  const _onFocus = () => {
+    console.log("_onFocus")
+
+    if (isListShown) {
+      setIsListShown(false)
+    } else {
+      setIsListShown(true)
+    }
+  }
+
+  React.useEffect(() => {
+    setDataDropdown(data)
+  }, [])
+
+  React.useEffect(() => {
+    console.log("item selected")
+    // setSelectedItem(itemSelected)
+    if (typeof itemSelected !== 'undefined') {
+      dataDropdown.map((item, index) => {
+        if (itemSelected.value === item.value) {
+          setSelectedItem(item)
+          indexSelected = index
+        }
+      })
+    }
+  }, [itemSelected])
+
+  React.useEffect(() => {
+    console.log("index selected")
+    // setSelectedItem(itemSelected)
+    if (typeof indexSelected !== 'undefined') {
+      dataDropdown.map((item, index) => {
+        if (indexSelected === index) {
+          setSelectedItem(item)
+        }
+      })
+    }
+  }, [indexSelected])
+
+  const _selectThisItem = (selected) => {
+    console.log("selected: ", selected)
+    setIsListShown(false)
+    setSelectedItem(selected)
+    input.current.blur()
+    onItemSelected(selected)
+  }
+
+  return (
+    <View style={[{
+      width: '90%',
+      margin: 4,
+      zIndex: 1000,
```

Comment:
This would be best if converted into a prop. This is an application-specific style that might not work for everyone.

Reply:
Alright, it was moved to props with default value. Thanks.